bokeh/io.py

```diff
@@ -346,6 +356,47 @@ def _save_helper(obj, filename, resources, title, validate):
         with io.open(filename, "w", encoding="utf-8") as f:
             f.write(decode_utf8(html))
 
+def savepng(obj, filename=None, state=None, validate=True):
+    if state is None:
+        state = _state
+
+    if filename is None and state.png:
+        filename = state.png['filename']
+
+    if filename is None:
+        raise RuntimeError("savepng() called but no filename was supplied and output_png(...) was never called, nothing saved")
+
+    with _ModelInDocument(obj):
+        if isinstance(obj, Component):
+            doc = obj.document
+        elif isinstance(obj, Document):
+            doc = obj
+        else:
+            raise RuntimeError("Unable to save object of type '%s'" % type(obj))
+
+        if validate:
+            doc.validate()
```

Comment:
the code from `with _ModelInDocument` through `validate()` could probably be factored out since it's in `_save_helper` as well